index.js

```diff
@@ -212,15 +212,15 @@ var compile = function(schema, cache, root, reporter, opts) {
     }
 
     if (node.format && fmts[node.format]) {
-      if (type !== 'string' && formats[node.format]) validate('if (%s) {', types.string(name))
+      if (type !== 'string' && fmts[node.format]) validate('if (%s) {', types.string(name))
```

Comment:
`fmts[node.format]` will always be truthy since it's checked in the outer `if`

Reply:
that's good observation, but does not affect the fix. the only difference between `formats` and `fmts` is that `fmts` has user supplied custom formats and built-ins.

Reply:
I meant that you can remove that part alltogether, since it will always be true:

```js
if (type !== 'string') validate('if (%s) {', types.string(name))
```